src/screens/Inventory/PurchaseRequest.tsx

```diff
@@ -73,15 +76,27 @@ function PurchaseRequest(props: PurchaseRequestsProps) {
   }
 
   const handleSubmit = (purchaseRequest: PurchaseRequestRecord, approved: boolean) => {
-    reviewPurchaseRequest(purchaseRequest, approved, currentUserId);
-    history.goBack();
+    if (!isOnline) {
+      setShowSnackbar(true);
+      // 5 second delay to reset back so it can be shown again.
+      setTimeout(function () {
+        setShowSnackbar(false);
+      }, 5000);
+    } else {
+      reviewPurchaseRequest(purchaseRequest, approved, currentUserId);
+      history.goBack();
+    }
```

Comment:
I don't think we should use `isOnline` here as the indicator when seeing if the purchase request has an offlineID seems to be the real limiting factor. We can execute the `reviewPurchaseRequest` so long as the `purchaseRequest` doesn't have an offline ID.

With the current workflow, the User isn't able to approve any purchase requests while offline, when I think we really only need to prohibit them from reviewing their own purchase request that they might've made when they were offline

Reply:
Switched to only block approvals for requests with offline ids! Good catch.